src/components/operations/operation-details/ko/runtime/operation-details.ts

```diff
@@ -194,7 +194,7 @@ export class OperationDetails {
 
         let associatedAuthServers: AuthorizationServer[];
         if (api.authenticationSettings?.oAuth2AuthenticationSettings.length > 0) {
-            associatedAuthServers = api.authenticationSettings?.oAuth2AuthenticationSettings.map(x => new AuthorizationServer(x.authorizationServer));
+            associatedAuthServers = await this.oauthService.getOauthServers(api.id);
```

Comment:
there is a small issue here, lets say you have both kind of auth servers. openid and oauth2, on the line 197 and 201, you are assigning to the associatedAuthServers arrays. So you are overriding the previous values. at the end, you will only have openid servers because you fetch it the latest.

Instead of overriding, you should append the new values.
associatedAuthServers.push(...oauth2Servers)
associatedAuthServers.push(...openidServers)

Reply:
You can't have both of them (there are validations for this on the backend).
I changed the second one to `if else` since there is no point to check if we have openID if we already have oAuth2, thanks for noticing.